server/controllers/systemCtrl.test.js

```diff
@@ -0,0 +1,23 @@
+describe('Not found', () => {
+  test('Gets correct error code', done => {
+    jest.mock('../configuration', () => {
+      return { server: {} }
+    })
+    jest.mock('../api', () => {})
+    jest.mock('../adldapClient', () => {})
+
+    const systemCtrl = require('./systemCtrl')
```

Comment:
Kanske är det bara jag i teamet, men jag tror att det är bäst praxis att alltid använder `require()` globalt och i början av filen... ;-)

Reply:
När det kommer till ES6 modules gäller [Imports and exports must be at the top level](https://exploringjs.com/es6/ch_modules.html#_imports-and-exports-must-be-at-the-top-level)
och [Imports are hoisted](https://exploringjs.com/es6/ch_modules.html#_imports-are-hoisted)... 🥳

Reply:
Hmmm, men hur löser man då mockningen?

Reply:
Titta till exempel på det här: https://github.com/KTH/kth-node-cosmos-db/blob/master/lib/utils/CosmosClientWrapper.test.js#L8

Mockning "cosmosDb" aktiveras (rad 8), importeras från "__mocks__/cosmosDb.js" (rad 13) innan source-filen (rad 23), kanske undvikas (rad 20) och anpassas (rad 116, fungerar bara i samband med `jest.fn()` istället för `() => {}`)...